functions/transformations/difference.go

```diff
@@ -187,7 +187,7 @@ func (t *differenceTransformation) Process(id execute.DatasetID, tbl flux.Table)
 				d := differences[j]
 				switch c.Type {
 				case flux.TBool:
-					if err := arrow.AppendBools(builder, j, arrow.BoolSlice(cr.Bools(j), firstIdx, l)); err != nil {
+					if err := builder.AppendBools(j, arrow.BoolSlice(cr.Bools(j), firstIdx, l)); err != nil {
```

Comment:
I'm not sure how this got in originally, but it's a memory leak. You have to release the slice after creating it.

Reply:
will let this get fixed in a currently open PR for difference.